### PR TITLE
[DUOS-3023][risk=no] Index dataset on DAC approval change

### PR DESCRIPTION
**Pull request by rushtong**
### Addresses
https://broadworkbench.atlassian.net/browse/DUOS-3023

### Summary
* Index the updated dataset on DAC approval/rejection
* Minor test class cleanup for JUnit5 conventions

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/resources/DacResource.java [hunk 1]
```diff
@@ -254,6 +258,13 @@ public Response approveDataset(@Auth AuthUser authUser, @PathParam("dacId") Inte
         throw new BadRequestException("Invalid request payload");
       }
       Dataset updatedDataset = datasetService.approveDataset(dataset, user, payload.getApproval());
+      try (Response indexResponse = elasticSearchService.indexDataset(updatedDataset))  {
+        if (indexResponse.getStatus() >= Status.BAD_REQUEST.getStatusCode()) {
+          logWarn("Non-OK response when reindexing dataset with id: " + datasetId);
+        }
+      } catch (Exception e) {
+        logException("Exception re-indexing datasets from dataset id: " + datasetId, e);
+      }
```
Comment (fboulnois) [inline src/main/java/org/broadinstitute/consent/http/resources/DacResource.java:267]: Good catch, shall we add tests to cover these lines?

**Review by fboulnois — APPROVED** [review]
Question:

**Review by aarohinadkarni — APPROVED** [review]
👍

**Review by fboulnois — APPROVED** [review]
thanks 👍

**Pull request merged**